spaceR/spaceR/Controllers/SignupViewController.swift

```diff
@@ -8,48 +8,34 @@
 import UIKit
 
 class SignupViewController: UIViewController {
-    
+ 
     @IBOutlet private weak var userFirstName: UITextField!
     @IBOutlet private weak var userLastName: UITextField!
     @IBOutlet private weak var userEmail: UITextField!
     @IBOutlet private weak var userPassword: UITextField!
-    private lazy var signUpViewModel = SignupViewModel()
+    private lazy var signUpViewModel = SignupViewModel(delegate: self)
     
     override func viewDidLoad() {
         super.viewDidLoad()
         self.navigationController?.navigationBar.isHidden = true
     }
     
     @IBAction private func signupButtonClicked(_ sender: Any) {
-        self.validateUserInputFromSignupFields() ? self.successfulSignup() : self.unsuccessfullSignup()
-    }
-    
-    private func validateUserInputFromSignupFields() -> Bool {
-        if let safeFirstName = self.userFirstName.text, !safeFirstName.isEmpty,
-           let safeLastName = self.userLastName.text, !safeLastName.isEmpty,
-           let safeEmail = self.userEmail.text, !safeEmail.isEmpty,
-           let safePassword = self.userPassword.text,
-           !safePassword.isEmpty {
-            setSignupModelAttributes(firstName: safeFirstName, lastName: safeLastName, email: safeEmail, password: safePassword)
-            return true
-        } else {
+        do {
+            try
+                signUpViewModel.validateUserInput(userFirstName: userFirstName.text,
+                                          userLastName: userLastName.text,
+                                          userEmail: userEmail.text,
+                                          userPassword: userPassword.text) ? self.successfulSignup() : self.unsuccessfullSignup()
```

Comment:
Same as above, instead of `signUpViewModel.validateUserInput` returning a `bool` value why not let your `ViewModel` do the evaluation and then call one of your delegate functions?